### PR TITLE
Enhancements from Nick W

### PR DESCRIPTION
**Pull request by orerocks**
- Level input data by removing mean data value (`"subtract_mean": true,`)
- Apply new uncertainties: (`"new_uncert": [0,5],` where first entry is % (0-1) and second is minimum floor)
- add dask parameters
- Allow stop after MVI-C (`"inversion_type":"MVI",` versus `"inversion_type":"MVIS",`)
- Enable start and reference models as either vector or inputs (`"model_start": [0,0,0],` or `"model_ref": 1e-4,`). If scalar then it assumes it is an amplitude in the direction of the inducing field.
- Add input switch for plotting graphics (`"show_graphics": true`)
- Plot a slice through the global mesh

### CODE REVIEW

**Comment by orerocks** [comment]
I avoided anything to do with ftmg by checking for `ubc_grav` or `ubc_mag` data types.

**Comment by orerocks** [comment]
- Add `max_IRLS_iter` as a JSON input parameter. Set defaults to 0 for MVI-C & not-sparse, and 20 for MVI-S and sparse.

File: PF_inversion_20190907.py [hunk 1]
```diff
@@ -95,10 +98,33 @@
...
+        survey.dobs -= d0
+        print('Removed data mean: {0:.6g}'.format(d0))
+else:
+    subtract_mean = False
+
+# Update the specified data uncertainty
+if "new_uncert" in list(input_dict.keys()) and input_dict["data_type"].lower() in ['ubc_mag', 'ubc_grav']:
+    new_uncert = input_dict["new_uncert"]
+else:
+    new_uncert = [-1, -1]
```
Comment (fourndo) [inline PF_inversion_20190907.py]: Can we do `new_uncert = False` instead? Cleaner

File: PF_inversion_20190907.py [hunk 2]
```diff
@@ -95,10 +98,33 @@
...
+
+if len(new_uncert) == 2 and all(np.asarray(new_uncert) >= 0):
```
Comment (fourndo) [inline PF_inversion_20190907.py]: `if new_uncert:` then add an `assert` that new_uncert as len == 2

File: PF_inversion_20190907.py [hunk 3]
```diff
@@ -95,10 +98,33 @@
...
+    survey.std = np.maximum(abs(new_uncert[0]*survey.dobs),new_uncert[1])
+    
+elif survey.std is None:
+    survey.std = survey.dobs * 0 + 10 # Default
```
Comment (fourndo) [inline PF_inversion_20190907.py]: I would suggest just setting the default to 1

File: PF_inversion_20190907.py [hunk 4]
```diff
@@ -177,13 +213,25 @@
 else:
     alphas = [1, 1, 1, 1]
 
+if "model_start" in list(input_dict.keys()):
+    model_start = np.r_[input_dict["model_start"]]
+    assert model_start.shape[0] == 1 or model_start.shape[0] == 3, "Start model needs to be a scalar or 3 component vector"
+else:
+    model_start = 0
+
+if "model_ref" in list(input_dict.keys()):
```
Comment (fourndo) [inline PF_inversion_20190907.py]: Can we spell out `model_reference`

File: PF_inversion_20190907.py [hunk 5]
```diff
@@ -363,6 +424,18 @@ def createLocalMesh(rxLoc, ind_t):
 else:
     mesh = meshInput
 
+if show_graphics:
```
Comment (fourndo) [inline PF_inversion_20190907.py:427]: That's great, we can add more arguments to those variables later to control the plot.

File: PF_inversion_20190907.py [hunk 6]
```diff
@@ -556,38 +636,60 @@ def createLocalProb(meshLocal, survey_t, wrGlobal, ind):
...
+    plt.yscale('log')
+    
+    twin = axs.twinx()
+    twin.plot(saveDict.phi_m, 'k--', lw=2)
+    plt.autoscale(enable=True, axis='both', tight=True)        
+
+    axs.set_ylabel('$\phi_d$', size=16, rotation=0)
+    axs.set_xlabel('Iterations', size=14)
+    twin.set_ylabel('$\phi_m$', size=16, rotation=0)
+    plt.show(block=False)
```
Comment (fourndo) [inline PF_inversion_20190907.py:692]: Do you want to `savefig` here as well?

**Review by fourndo — APPROVED** [review]
Up to you to implement the changes. We can also keep on building from there.

File: PF_inversion_20190907.py (hunk 1 above)
Comment (orerocks) [reply PF_inversion_20190907.py]: How about ad18102? new_uncert isn't used anywhere else.

File: PF_inversion_20190907.py (hunk 6 above)
Comment (orerocks) [reply PF_inversion_20190907.py:692]: Done 24e83a6

File: PF_inversion_20190907.py (hunk 4 above)
Comment (orerocks) [reply PF_inversion_20190907.py]: Done 90124b5

**Pull request merged**